snf-cyclades-app/synnefo/logic/servers.py

```diff
@@ -218,6 +224,7 @@ def create_server(vm, nics, volumes, flavor, image, personality, password):
         'img_format': str(image['format']),
         'img_personality': json.dumps(personality),
         'img_properties': json.dumps(image['metadata']),
+        'auth_keys': public_key,
```

Comment:
Well, make sure that you end up with a `None` `public_key` if the user hasn't defined a `key_name` parameter and don't insert the `auth_keys` parameter if this is the case.